core/src/com/unciv/models/ruleset/unique/Conditionals.kt

```diff
@@ -99,6 +99,48 @@ object Conditionals {
             return compare(statReserve, lowerLimit * gameSpeedModifier, upperLimit * gameSpeedModifier)
         }
 
+        fun getCountableAmount(countable: String): Float? {
+            if (countable.toIntOrNull() != null) return countable.toFloat()
+            if (countable.toFloatOrNull() != null) return countable.toFloat()
+            if (countable == "year") return gameInfo!!.getYear(gameInfo!!.turns).toFloat()
+
+            val relevantStat = Stat.safeValueOf(countable)
+
+            if (relevantStat != null) {
+                if (relevantCity != null) {
+                    return relevantCity!!.getStatReserve(relevantStat).toFloat()
+                } else if (relevantStat in Stat.statsWithCivWideField) {
+                    return relevantCiv!!.getStatReserve(relevantStat).toFloat()
+                } else {
+                    return null
+                }
+            }
+
+            if (gameInfo!!.ruleset.tileResources.containsKey(countable))
+                return getResourceAmount(countable).toFloat()
+
+            return null
+        }
+
+        fun compareCountables(
+            first: String,
+            second: String,
+            compare: (first: Float, second: Float) -> Boolean): Boolean {
+
+            return if (getCountableAmount(first) != null && getCountableAmount(second) != null)
```

Comment:
more like
firstCountable = getCountableAmount(first)
secondCountable = getCountableAmount(second)

That way we only need to run the function once

Same for the rest of these